src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala

```diff
@@ -158,68 +227,106 @@ object HouseholdActor {
-            veh.manager = Some(self)
-            veh.spaceTime = SpaceTime(homeCoord.getX, homeCoord.getY, 0)
-            parkingManager ? ParkingInquiry(
-              homeCoord,
-              homeCoord,
-              "home",
-              AttributesOfIndividual.EMPTY,
-              NoNeed,
-              0,
-              0
-            ) flatMap {
-              case ParkingInquiryResponse(stall, _) =>
-                veh.useParkingStall(stall)
-                self ? ReleaseVehicleAndReply(veh)
+      case RoutingResponses(tick, routingResponses) =>
+        // Index the responsed by Id
+        val indexedResponses = routingResponses.map(resp => (resp.requestId -> resp)).toMap
+        // Create a passenger schedule for each CAV in the plan, split by passenger to be picked up
+        cavPassengerSchedules = cavPlans.map{ cavSchedule =>
+          val theLegsWithServiceRequest = cavSchedule.schedule.map{ serviceRequest =>
+            (indexedResponses(serviceRequest.routingRequestId.get).itineraries.head.legs.head.beamLeg,serviceRequest)
+          }
+          //TODO the split needs to happen before a positive result, not with the positive result, hmmmm
+          val splitByPickup = RandomUtils.multiSpan(theLegsWithServiceRequest)(_._2.tag != Pickup)
+          val passengerSchedulesByPickup = splitByPickup.map{ subLegsWithServiceRequest =>
+            var passengerSchedule = PassengerSchedule().addLegs(subLegsWithServiceRequest.map(_._1))
+            subLegsWithServiceRequest.filter(_._2.person.isDefined).map{ legWithPassenger =>
+              passengerSchedule = passengerSchedule.addPassenger(memberVehiclePersonIds(legWithPassenger._2.person.get),Seq(legWithPassenger._1))
             }
-          })
-          .map(_ => CompletionNotice(triggerId, Vector()))
-          .pipeTo(sender())
+            passengerSchedule
+          }
+          (cavSchedule.cav -> passengerSchedulesByPickup)
+        }.toMap
+        cavPassengerSchedules.foreach { cavAndSchedules =>
+          val nextSchedule = cavAndSchedules._2.head
+          if(nextSchedule.schedule.map(_._2.riders.size).sum == 0){
+            cavAndSchedules._1.driver.get ! ModifyPassengerSchedule(nextSchedule,tick)
+          }
+        }
+        val (_, triggerId) = releaseTickAndTriggerId()
+        completeInitialization(triggerId)
 
       case NotifyVehicleIdle(vId, whenWhere, _, _, _) =>
         val vehId = vId.asInstanceOf[Id[BeamVehicle]]
         vehicles(vehId).spaceTime = whenWhere
         log.debug("updated vehicle {} with location {}", vehId, whenWhere)
 
       case ReleaseVehicle(vehicle) =>
-        vehicle.unsetDriver()
-        availableVehicles = vehicle :: availableVehicles
-        log.debug("Vehicle {} is now available for anyone in household {}", vehicle.id, household.getId)
+        if(vehicle.beamVehicleType.automationLevel > 3){
+          //TODO update cav schedule and re-dispatch this vehicle if necessary
```

Comment:
Per a recent issue by Colin, I think the preference is to open a github issue instead of adding a TODO in the code. So, can we get rid of this nop, and open an issue to track work on this?